Source/ProgressionSystemRuntime/Public/Widgets/PSMenuWidget.h

```diff
@@ -0,0 +1,72 @@
+	 * @param AmountOfUnlockedPoints The number of images (unlocked-icon as images) to be displayed 
+	 * @param AmountOfLockedPoints The number of images (locked-icon as images) to be displayed
+	 */
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void AddImagesToHorizontalBox(float AmountOfUnlockedPoints, float AmountOfLockedPoints);
+
+	/**
+	 * Dynamically populates a Horizontal Box with images representing unlocked and locked progression icons.
+	* @param CreatedObjects - Handles of objects from Pool Manager
+	* @param AmountOfUnlockedPoints The number of images (unlocked-icon as images) to be displayed 
+	 * @param AmountOfLockedPoints The number of images (locked-icon as images) to be displayed
+	 */
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void OnTakeFromPoolCompleted(const TArray<FPoolObjectData>& CreatedObjects, float AmountOfUnlockedPoints, float AmountOfLockedPoints);
+
+	/**
+	* Sets the visibility of the background overlay and lock icon.
+	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
+	*/
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected, BindWidget))
+	void SetOverlayVisibility(ESlateVisibility VisibilitySlate);
+
+	/*********************************************************************************************
+	 * Protected functions
+	 ********************************************************************************************* */
+protected:
+	// Horizontal Box widget for storing stars
+	UPROPERTY(EditDefaultsOnly, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UHorizontalBox> HorizontalBox = nullptr;
+
+	/** Background overlay for tint effect */
+	UPROPERTY(EditDefaultsOnly, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UImage> PSCBackgroundOverlay = nullptr;
+
+	/** Background overlay lock icon */
+	UPROPERTY(EditDefaultsOnly, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UImage> PSCBackgroundIconLock = nullptr;
+
+	/** Array of pool handlers which should be released */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Pool Widget Handlers"))
+	TArray<FPoolObjectHandle> PoolWidgetHandlersInternal;
+
+	/** Called after the underlying slate widget is constructed.
+	 * May be called multiple times due to adding and removing from the hierarchy. */
+	virtual void NativeConstruct() override;
+	
+	/** Updates star images icon to locked/unlocked according to input amounnt
+	 * @param CreatedData Object received from Pool Manager which contains the reference to Start Widget 
+	 * @param AmountOfUnlockedStars Amount of icons to be switched to Unlocked stars.
+	 * @param AmountOfLockedStars Amount of icons to be switched to Locked stars.
+	 */
+	void UpdateStarImages(const FPoolObjectData& CreatedData, float AmountOfUnlockedStars, float AmountOfLockedStars);
```

Comment:
please do it `UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))`

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/7bf3b056eb9837a926d7e4864951be7385a9d5e2